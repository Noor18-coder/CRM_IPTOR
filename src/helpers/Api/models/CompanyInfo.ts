// export interface CompanyInfoItem {
//   companyShortName: string;
//   name: string;
//   addressLine1: string;
//   addressLine2: string;
//   addressLine3: string;
//   postalCode: string;
//   partyIdentification: string;
//   phoneNumber: string;
//   faxNumber: string;
//   udPhoneNumber: string;
//   udFaxNumber: string;
//   registrationNo: string;
//   vatRegistrationNo: string;
//   systemCountry: string;
//   systemLanguage: string;
//   systemCurrency: string;
//   goodsAddressNo: number;
//   palletRegistrationNo: number;
//   county: string;
//   state: string;
//   taxJurisdictionCode: string;
//   reportingCurrency: string;
//   eMailAddress: string;
//   homePage: string;
//   ibanAccountNo: string;
//   bankIdentificationCode: string;
//   bankName: string;
//   bankAddress: string;
//   swiftAddress: string;
//   businessIdentityText: string;
//   vatRegisterOffice: string;
//   recordFound: boolean;  
//   environmentId: string;
//   company: string;
// }

export interface CompanyInfoItem {
  companyCode : string,
  companyShortName : string,
  name : string, 
  selected: boolean
}; 

export interface CompanyInfoRes {
  items:CompanyInfoItem[]
}

export interface CompanyInfoItemResponse {
  IptorAPI: string;
  items: CompanyInfoItem[];
  id: string;
}