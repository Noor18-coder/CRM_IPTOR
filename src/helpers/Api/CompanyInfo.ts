import axios from 'axios';
import { get } from 'lodash';
import { CompanyInfoItemResponse } from './models';
import { ApiRequest } from './ApiRequest';

export class CompanyInfo {
  /** API Method */
  private static apiMethod = 'environmentCompanies.get';

  /**
   * Helper function to fetch sales order list
   * @param params user id
   * @returns information about user
   */
  static async get(): Promise<CompanyInfoItemResponse> {
    const requestData = new ApiRequest(this.apiMethod);
    const response = await axios.post<CompanyInfoItemResponse>('/api/service', requestData);
    return get(response, 'data.data', []);
  }
}
