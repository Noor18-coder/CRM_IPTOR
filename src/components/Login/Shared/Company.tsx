import React from 'react';
import { CompanyInfoItem } from '../../../helpers/Api/models';
import VectorImg from '../../../assets/images/check_circle.svg';

export interface Props {
    companies: CompanyInfoItem[],
    doClick: (key: string) => void
}

export const Company: React.FC<Props> = ({
    companies,
    doClick
}) => {
    return (
        <ul style={{ margin: 0, padding: 0 }}>
            {
                companies.map((obj) => {
                    return (
                        <div className={'company-container'} onClick={() => doClick(obj.companyCode)} key={obj.companyCode}>
                            <div className={'center'}>
                                <div className={'test'}>{obj.name}</div>
                                <img className={'company-selection-img'} style={{ display: obj.selected ? 'block' : 'none' }} src={VectorImg}></img>
                            </div>
                        </div>
                    )
                })
            }
        </ul>
    )
};