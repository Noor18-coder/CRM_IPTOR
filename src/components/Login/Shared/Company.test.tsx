import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Company } from './Company';
import { CompanyInfoMock } from '../../../mocks/CompanyInfo.mock';

const companies = CompanyInfoMock.getCompaniesConstVal();
const doClick = jest.fn();

describe('[Login Shared] Company', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Company companies={companies} doClick={doClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call doClick on click', () => {
    const component = mount(<Company companies={companies} doClick={doClick} />);

    const li = component.find('li.action-icon');
    li.at(0).simulate('click');

    expect(doClick.mock.calls.length).toBeGreaterThan(0);
  });

  it('should call doClick on keypress', () => {
    const component = mount(<Company companies={companies} doClick={doClick} />);

    const li = component.find('li.action-icon');
    li.at(0).simulate('keydown', { key: 'Enter' });

    expect(doClick.mock.calls.length).toBeGreaterThan(0);
  });
});
