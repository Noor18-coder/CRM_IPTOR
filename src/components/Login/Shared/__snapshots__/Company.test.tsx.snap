// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Login Shared] Company should renders correctly 1`] = `
<ul
  className="companys-list-item"
>
  <li
    className="action-icon"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
  >
    <div
      className="company-container"
    >
      <div
        className="center"
      >
        <div
          className="test"
        >
          Steuber - Abernathy
        </div>
        <img
          alt="company"
          className="company-selection-img"
          src="check_circle.svg"
          style={
            Object {
              "display": "none",
            }
          }
          title="company"
        />
      </div>
    </div>
  </li>
  <li
    className="action-icon"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
  >
    <div
      className="company-container"
    >
      <div
        className="center"
      >
        <div
          className="test"
        >
          Steuber - Abernathy
        </div>
        <img
          alt="company"
          className="company-selection-img"
          src="check_circle.svg"
          style={
            Object {
              "display": "block",
            }
          }
          title="company"
        />
      </div>
    </div>
  </li>
</ul>
`;
